8.0 port sale sourced by line

yvaucher opened the pull request:
Now allows to set a warehouse on sale order line

Added a test for sale order with multiple warehouses sources

yvaucher committed:
reactivate module sale_sourced_by_line

yvaucher committed:
port sale_sourced_by_line to 8.0

Now allows to set a warehouse on sale order line

Added a test for sale order with multiple warehouses sources

yvaucher committed:
move warehouse_id above route_id field in view

jgrandguillaume commented on sale_sourced_by_line/view/sale_view.xml:
```diff
@@ -17,9 +17,9 @@
-                <xpath expr="//field[@name='order_line']/form/group/group/field[@name='product_id']"
-                  position="after">
-                    <field name="location_id" domain="[('usage', '!=', 'view')]"/>
+                <xpath expr="//field[@name='order_line']/form/group/group/field[@name='route_id']"
```
This doesn't work because route_id isn't define here on so lines. replacing 'route_id' by 'price_unit' works.

jgrandguillaume commented:
Hi,

Thanks for porting this one ! My first comments:
- The view commented in-line above
- In openerp.py, at the end, it is MTO and Drop-shipping (not MTS and..)
- Setting up a default WH on SO doesn't fulfill by default the Source WH in the SO Line

Otherwise it's working properly, trying with MTO and dropshipping route is also working. Great !

Regards,

Joël

lepistone commented on sale_sourced_by_line/__openerp__.py:
```diff
@@ -20,6 +20,7 @@
 
 {'name': 'Sale Sourced by Line',
+ 'summary': 'Multiple warehouse source locations for Sale order',
  'version': '0.1',
```
I think we can bump the version, maybe 1.0?

yvaucher replied:
In fact it becomes the version 8.0.0.0.1 as odoo concatenate the odoo version

So I don't usually bump version when porting

lepistone replied:
Yes, but that is not a straight port: the module has really been changed. I'd bump in that case.

yvaucher committed:
improve warehouse help tooltip

yvaucher commented:
@jgrandguillaume fixed the view and fixed typos in `__openerp__`.py

For the WH, if it is not defined on the line, _prepare_order_line_procurement will use order.warehouse_id. I don't think duplicating the value is necessary. 

I added a better help tooltip on line warehouse_id

yvaucher committed:
Add POT file

yvaucher committed:
bump version

yvaucher commented:
@lepistone bumped the version as you insist :smile_cat:

yvaucher committed:
put default warehouse_id of sale order in warehouse_id of sale order line

yvaucher commented:
@jgrandguillaume I setted the default warehouse_id on sale_order_line via xml view.

lepistone commented:
@yvaucher the build fails. Would you like me to take care of that?

lepistone committed:
try to test sale_sourced_by_line separately

lepistone commented:
@yvaucher we're green!

yvaucher commented:
@lepistone great! thanks for spliting the tests

lepistone commented:
:+1:

jgrandguillaume commented:
:+1:
